database-tech-it-easy.sql

```diff
@@ -0,0 +1,236 @@
+-- DROP tables if they already exist
+DROP TABLE IF EXISTS Users;
+DROP TABLE IF EXISTS Products CASCADE;
+DROP TABLE IF EXISTS ProductSaleDates CASCADE;
+DROP TABLE IF EXISTS Remotes CASCADE;
+DROP TABLE IF EXISTS CiModules CASCADE;
+DROP TABLE IF EXISTS WallBrackets CASCADE;
+DROP TABLE IF EXISTS Televisions CASCADE;
+DROP TABLE IF EXISTS TelevisionsWallBrackets CASCADE;
+
+--CREATE tables
+CREATE TABLE Users (
+    username VARCHAR(58) PRIMARY KEY,
+    password VARCHAR(150) NOT NULL,
+    email VARCHAR(150),
+    authorizationRole VARCHAR(24) NOT NULL
+);
+
+
+CREATE TABLE Products (
+    id SERIAL PRIMARY KEY,
+    name VARCHAR(58) NOT NULL,
+    brand VARCHAR(58),
+    type VARCHAR(36),
+    price DOUBLE PRECISION NOT NULL CHECK (price >= 0),
+    amountInStock INT DEFAULT 0,
+    amountSold INT DEFAULT 0
+);
+
+CREATE TABLE ProductSaleDates (
+    saleDate DATE NOT NULL,
+    product_id BIGINT,
+    CONSTRAINT fk_product FOREIGN KEY (product_id) REFERENCES Products(id) ON DELETE CASCADE
+);
+
+
+CREATE TABLE Remotes (
+    id BIGINT PRIMARY KEY REFERENCES Products(id) ON DELETE CASCADE,
+    batteryType VARCHAR(24),
+    isSmart BOOLEAN
+);
+
+
+CREATE TABLE CiModules (
+    id BIGINT PRIMARY KEY REFERENCES Products(id) ON DELETE CASCADE,
```

Comment:
je verwijst een paar zoals hier naar de tabel product en dan product(id). Is het dan geen Foreign key ipv primary? ik weet het niet zeker hoor misschien dat dit ook kan